crates/router/src/core/webhooks/incoming.rs

```diff
@@ -52,6 +52,58 @@ use crate::{
 #[cfg(feature = "payouts")]
 use crate::{core::payouts, types::storage::PayoutAttemptUpdate};
 
+pub fn webhook_not_found_error_handler(
+    req: &actix_web::HttpRequest,
+    connector: ConnectorEnum,
+    connector_name: String,
+    body: actix_web::web::Bytes,
+    error: error_stack::Report<errors::ApiErrorResponse>,
+) -> errors::RouterResult<(
+    services::ApplicationResponse<serde_json::Value>,
+    WebhookResponseTracker,
+    serde_json::Value,
+)> {
+    let connector_enum = api_models::connector_enums::Connector::from_str(&connector_name)
+        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+            field_name: "connector",
+        })
+        .attach_printable_lazy(|| format!("unable to parse connector name {connector_name:?}"))?;
+
+    let is_not_found_error = helper_utils::is_webhook_not_found_error(error.current_context());
+
+    // bypass error in webhook if connector supports it
+    if connector_enum.should_connector_bypass_error_if_webhook_not_found() && is_not_found_error {
+        logger::error!(?error, "Error in incoming webhooks core");
+
+        // append custom header to response
+        let header_name = actix_web::http::header::HeaderName::from_static("x-http-code");
+        let header_value = actix_web::http::header::HeaderValue::from_static("404");
+        let mut headers = req.headers().clone();
+        headers.append(header_name, header_value);
+
+        let request_details = IncomingWebhookRequestDetails {
+            method: req.method().clone(),
+            uri: req.uri().clone(),
+            headers: &headers,
+            query_params: req.query_string().to_string(),
+            body: &body,
+        };
+
+        let response = connector
+            .get_webhook_api_response(&request_details)
+            .switch()
+            .attach_printable("Could not get incoming webhook api response from connector")?;
+
+        Ok((
+            response,
+            WebhookResponseTracker::NoEffect,
+            serde_json::Value::Null,
+        ))
+    } else {
```

Comment:
This seems unnecessarily complicated in my opinion, took me some time to wrap my head around it:

- At this point, we already know that there's an error, we know the type of error, and that we should be ignoring object not found errors for this specific connector.
- Yet, we modify the request headers, construct an incoming webhook request (again) and then call `get_webhook_api_response()`.
- In this specific case, we are modifying request headers to indicate that it was a object not found error.

I believe we also have a fair idea about where in the incoming webhooks core code we'd get an object not found error. Given that we know where we'd get an object not found error, can't we handle it suitably in the incoming webhooks core itself and return a connector-specific response?

For generating a connector-specific response, we can do one of two things:
- Add a new trait method along the lines of `get_webhook_object_not_found_api_response()`.
- Add a new parameter to the `get_webhook_api_response()` that either indicates the type of error (if applicable), or specifies whether it is an object not found error, and the connector code can choose to ignore the error or not (and generate a suitable response) based on this newly introduced parameter.

Do let me know if I'm missing something here.